libbeat/docs/outputs/output-logstash.asciidoc

```diff
@@ -0,0 +1,397 @@
+Want to use <<filebeat-modules,{beatname_uc} modules>> with {ls}? You need to do
+some extra setup. For more information, see
+{logstash-ref}/filebeat-modules.html[Working with {beatname_uc} modules].
+endif::[]
+
+ifndef::win-only[]
+ifndef::apm-server[]
+ 
+include::{libbeat-dir}/docs/step-test-config.asciidoc[]
+
+endif::apm-server[]
+ifdef::apm-server[]
+
+include::./step-test-config.asciidoc[]
+
+endif::apm-server[]
+endif::win-only[]
+
+ifdef::win-only[]
+
+TIP: To test your configuration file, change to the directory where the {beatname_uc}
+binary is installed, and run {beatname_uc} in the foreground with the following
+options specified: +.\winlogbeat.exe test config -c .\winlogbeat.yml -e+.
+
+endif::win-only[]
+
+// end::shared-logstash-config[]
+
+==== Accessing metadata fields
+
+Every event sent to Logstash contains the following metadata fields that you can
+use in Logstash for indexing and filtering:
+
+ifndef::apm-server[]
+["source","json",subs="attributes"]
+------------------------------------------------------------------------------
+{
+    ...
+    "@metadata": { <1>
+      "beat": "{beat_default_index_prefix}", <2>
+      "version": "{stack-version}" <3>
+    }
+}
+------------------------------------------------------------------------------
+<1> {beatname_uc} uses the `@metadata` field to send metadata to Logstash. See the
+{logstash-ref}/event-dependent-configuration.html#metadata[Logstash documentation]
+for more about the `@metadata` field.
+<2> The default is {beat_default_index_prefix}. To change this value, set the
+<<logstash-index,`index`>> option in the {beatname_uc} config file.
+<3> The current version of {beatname_uc}.
+endif::[]
+
+ifdef::apm-server[]
+["source","json",subs="attributes"]
+------------------------------------------------------------------------------
+{
+    ...
+    "@metadata": { <1>
+      "beat": "{beat_default_index_prefix}", <2>
+      "pipeline":"apm", <3>
+      "version": "{stack-version}" <4>
+    }
+}
+------------------------------------------------------------------------------
+<1> {beatname_uc} uses the `@metadata` field to send metadata to Logstash. See the
+{logstash-ref}/event-dependent-configuration.html#metadata[Logstash documentation]
+for more about the `@metadata` field.
+<2> The default is {beat_default_index_prefix}. To change this value, set the
+<<logstash-index,`index`>> option in the {beatname_uc} config file.
+<3> The default pipeline configuration: `apm`. Additional pipelines can be enabled
+with a {logstash-ref}/use-ingest-pipelines.html[Logstash pipeline config].
+<4> The current version of {beatname_uc}.
+endif::[]
+
+You can access this metadata from within the Logstash config file to set values
+dynamically based on the contents of the metadata.
+
+For example, the following Logstash configuration file tells
+Logstash to use the index reported by {beatname_uc} for indexing events
+into Elasticsearch:
+
+ifndef::apm-server[]
+[source,logstash]
```

Comment:
Ohh awesome! I forgot these updates were in the outputconfig file 🤦‍♂. Are you planning on backporting this PR? If so, there's no reason for me to backport https://github.com/elastic/beats/pull/13596.

Reply:
@bmorelli25 I will backport this to 7.4 but not earlier.

Reply:
@bmorelli25 Actually, I will backport #13596 today because I want to have a good diff of the master and 7.4 branches on outputconfig.asciidoc before backporting this PR (will be easier to see changes). Also, there are some additional changes in 13596 PR that you probably want backported, too, but might not be reflected here.